docs/architecture/pk-4.md

```diff
@@ -0,0 +1,352 @@
+
+This includes both direct mentions (where a type name, constant name or function name is mentioned), and indirect mentions (e.g. “verify\_ext”, “context has been set up”).
+
+ACTION: update unit tests to validate the new interfaces.
+
+#### Remove RSA-ALT
+
+ACTION: Remove the option `MBEDTLS_PK_RSA_ALT_SUPPORT` and all code guarded by it, as well as `MBEDTLS_PK_RSA_ALT`.
+
+### Documentation updates
+
+ACTION: update the PSA transition guide.
+
+ACTION: write changelog entries.
+
+## Open questions
+
+### Parsing and writing
+
+#### Remove file parsing functions?
+
+Should we remove the file parsing functions `mbedtls_pk_parse_keyfile()` and `mbedtls_pk_parse_public_keyfile()`? They look misplaced in a library that generally doesn't access files. But it isn't really difficult to keep them.
+
+These functions are used in test code and sample programs.
+
+### Resource management
+
+#### Access the underlying PSA key of a PK context
+
+Should we provide a function to access the underlying PSA key of a PSA context, if there is one?
+
+## Later tasks
+
+The tasks described in this section do not need to be done before the 1.0/4.0 release. This section is incomplete.
+
+### Missing functionality
+
+#### Remove limitations of `mbedtls_pk_wrap_psa()`
+
+Remove the limitations of wrapped keys:
+
+* Implement verify for wrapped RSA keys.
+* Implement check-pair for wrapped RSA keys.
+
+### Migrate library and test code
+
+#### Update terminology from “opaque” to “wrapped”
+
+Update the library code to change the obsolete terminology “opaque” (as in e.g. `MBEDTLS_PK_OPAQUE`) to “wrapped”.
+
+Originally, PK contexts could wrap a PSA key in order to support non-exportable keys kept in a secure partition, using a PSA implementation with client-server isolation. Thus the useful purpose of wrapping a PSA key was to make the key opaque, hence the name. In TF-PSA-Crypto, we plan to evolve to making PK keys contain a PSA key in more situations. What will matter is whether the underlying PSA key is owned by the PK context (and destroyed when the context is freed), or not (thus surviving when the context is freed). A key owned by the PK context could still potentially be opaque. Hence the distinction should be between _wrapped_ and _owned_ PSA keys.
+
+#### Retire `MBEDTLS_PK_ECDSA`
+
+It is no longer possible to construct a PK object with the legacy type `MBEDTLS_PK_ECDSA`. So get rid of the code that handles it, and remove `MBEDTLS_PK_ECDSA`.
+
+#### Replace all uses of `mbedtls_pk_type_t`, `mbedtls_pk_get_type()` and `mbedtls_pk_can_do()` in X.509 library and test code
+
+#### Replace all uses of `mbedtls_pk_type_t`, `mbedtls_pk_get_type()` and `mbedtls_pk_can_do()` in TLS library and test code
+
+#### Get rid of `mbedtls_pk_can_do()`
+
+#### Get rid of `mbedtls_pk_info`
+
+Dispatch based on an enum rather than a method table. This simplifies the code.
+
+#### Make `mbedtls_pk_type_t` internal
```

Comment:
Note: so you're making a distinction between private (not part of the public API, but accessible to X.509 and TLS) and internal (only used in `pk*.c` and possibly PK unit tests). Ok, internal dictionary updated.

Reply:
I tried to follow the same terminology as https://github.com/Mbed-TLS/TF-PSA-Crypto/blob/79d10321efd0c50ccad9bf37f0223c295cdfa1ba/docs/architecture/0e-plans.md